hexl/fft/fft.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2020-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include "hexl/fft/fft.hpp"
+
+#include "hexl/fft/fft-native.hpp"
+#include "hexl/logging/logging.hpp"
+
+namespace intel {
+namespace hexl {
+
+// AllocatorStrategyPtr mallocStrategy = AllocatorStrategyPtr(new
+// MallocStrategy);
+
+FFT::FFT(uint64_t degree, double* in_scalar,
+         std::shared_ptr<AllocatorBase> alloc_ptr)
+    : m_degree(degree),
+      scalar(in_scalar),
+      m_alloc(alloc_ptr),
+      m_aligned_alloc(AlignedAllocator<double, 64>(m_alloc)),
+      m_complex_roots_of_unity(m_aligned_alloc) {
+  HEXL_CHECK(IsPowerOfTwo(degree),
+             "degree " << degree << " is not a power of 2");
+  HEXL_CHECK(degree > 8, "degree should be bigger than 8");
+
+  m_degree_bits = Log2(m_degree);
+  ComputeComplexRootsOfUnity();
+
+  if (scalar != nullptr) {
+    scale = *scalar / static_cast<double>(degree);
+    inv_scale = static_cast<double>(1.0) / *scalar;
```

Comment:
Please look through the code and sort these out. Do not expect the reviewer to find them all :D
```suggestion
    inv_scale = 1.0 / *scalar;
```